pkg/term/term_windows.go

```diff
@@ -119,20 +168,30 @@ func MakeRaw(fd uintptr) (*State, error) {
 	mode := state.mode
 
 	// Disable these modes
-	mode &^= winconsole.ENABLE_ECHO_INPUT
-	mode &^= winconsole.ENABLE_LINE_INPUT
-	mode &^= winconsole.ENABLE_MOUSE_INPUT
-	mode &^= winconsole.ENABLE_WINDOW_INPUT
-	mode &^= winconsole.ENABLE_PROCESSED_INPUT
+	mode &^= windows.ENABLE_ECHO_INPUT
+	mode &^= windows.ENABLE_LINE_INPUT
+	mode &^= windows.ENABLE_MOUSE_INPUT
+	mode &^= windows.ENABLE_WINDOW_INPUT
+	mode &^= windows.ENABLE_PROCESSED_INPUT
 
 	// Enable these modes
-	mode |= winconsole.ENABLE_EXTENDED_FLAGS
-	mode |= winconsole.ENABLE_INSERT_MODE
-	mode |= winconsole.ENABLE_QUICK_EDIT_MODE
+	mode |= windows.ENABLE_EXTENDED_FLAGS
+	mode |= windows.ENABLE_INSERT_MODE
+	mode |= windows.ENABLE_QUICK_EDIT_MODE
 
-	err = winconsole.SetConsoleMode(fd, mode)
-	if err != nil {
+	if err := windows.SetConsoleMode(fd, mode); err != nil {
 		return nil, err
 	}
 	return state, nil
 }
+
+func restoreAtInterrupt(fd uintptr, state *State) {
+	sigchan := make(chan os.Signal, 1)
+	signal.Notify(sigchan, os.Interrupt)
+
+	go func() {
+		_ = <-sigchan
+		RestoreTerminal(fd, state)
+		os.Exit(0)
```

Comment:
@tiborvass I'm not sure I understand your test. Admittedly, I may have misunderstood the signal handling as well.

In your test, since you've not pressed Enter, your shell remains in control and handles the ^C locally. Docker does not get involved in this case. The intent of the code, on the other hand, is to capture a signaled exit and restore the terminal state prior to exit. That said, I suspect the code here is weak and fails to restore all that it should (e.g., colors). However, the prior code did not even attempt to catch the signal. I feel this to be a step in the right direction.